apps/condo/domains/common/utils/text.spec.js

```diff
@@ -0,0 +1,43 @@
+const { normalizeText } = require('./text')
+
+describe('normalizeText()', () => {
+    test('empty text', () => {
+        expect(normalizeText('')).toBeUndefined()
+    })
+
+    test('returns normalized text as is', () => {
+        expect(normalizeText('lorem ipsum')).toEqual('lorem ipsum')
+        expect(normalizeText('lorem   ipsum')).toEqual('lorem ipsum')
+    })
+
+    test('removes leading and trailing spaces', () => {
+        expect(normalizeText('            lorem ipsum dolor           ')).toEqual('lorem ipsum dolor')
+        expect(normalizeText('  lorem   \n       ipsum   \n    dolor')).toEqual('lorem\nipsum\ndolor')
+    })
+
+    test('squashes sequential spaces into one', () => {
+        expect(normalizeText('lorem        ipsum    dolor')).toEqual('lorem ipsum dolor')
+    })
+
+    test('squashes sequential non-breaking spaces into one', () => {
+        expect(normalizeText('lorem   ipsum dolor')).toEqual('lorem ipsum dolor')
+    })
+
+    test('squashes sequential blank lines into one', () => {
+        expect(normalizeText('123\n\n\n\n123\n\n\n456'))
+            .toEqual('123\n\n123\n\n456')
+    })
+
+    test('removes extra spaces around punctuations', () => {
+        expect(normalizeText('123c .  sad ...')).toEqual('123c. sad...')
+        expect(normalizeText('например ,мама мыла раму . срочно!!!')).toEqual('например, мама мыла раму. срочно!!!')
+        expect(normalizeText('abc    ,  asd123 :123 \n 321 .    test')).toEqual('abc, asd123: 123\n321. test')
+    })
+
+    test('removes trailing and leading spaces inside quotes', () => {
```

Comment:
what about single quotes and other language-specific quotes?

Reply:
It will get tricky, because single quotes, as @nomerdvadcatpyat mentioned, can be used as apostrophes.
Maybe not cover this case?